pkg/agent/controller/networkpolicy/reconciler.go

```diff
@@ -866,6 +1007,23 @@ func (r *reconciler) Forget(ruleID string) error {
 	return nil
 }
 
+func (r *reconciler) getIGMPGroupAddress(rule *CompletedRule) []string {
+	groupAddresses := []string{}
+	if len(rule.Services) > 0 && (rule.Services[0].Protocol != nil) &&
+		(*rule.Services[0].Protocol == v1beta2.ProtocolIGMP) {
+		for _, service := range rule.Services {
+			if service.GroupAddress != "" {
+				groupAddresses = append(groupAddresses, service.GroupAddress)
+			}
+		}
+		if len(groupAddresses) == 0 && rule.Direction == v1beta2.DirectionIn {
+			groupAddresses = append(groupAddresses, types.McastAllHosts.String())
```

Comment:
this doesn't look correct, it's deciding the meaning of a service based on content of other services.
```
protocols:
- igmp:
    groupAddress: ""
- igmp:
    groupAddress: "224.0.0.5"   
```
For example, the first protocol would mean all multicast hosts when the second protocol exists but mean nothing when the second protocol doesn't exist.
I'm not sure what you intend to implement, we need to add such special cases to PR or issue decription to be clear: unset value, empty string, etc.

Reply:
changed to use *rule.Services[0].Protocol == v1beta2.ProtocolIGMP